### PR TITLE
Add method to visit annotation attachments in functions

### LINKED ISSUE #37381: Annotation attachments are not visited under functions in NodeFinder
**Description:**
See the comment https://github.com/ballerina-platform/ballerina-lang/issues/36351#issuecomment-1214840774

### PR DESCRIPTION
**Pull request by DamithaSenevirathne**
Fixes #37381

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFinder.java [hunk 1]
```diff
@@ -1264,6 +1264,10 @@ public void visit(BLangClassDefinition classDefinition) {
         lookupNodes(classDefinition.fields);
         lookupNodes(classDefinition.referencedFields);
         lookupNode(classDefinition.initFunction);
+        for (BLangFunction method : classDefinition.functions) {
```
Comment (IMS94) [inline compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFinder.java:1267]: Isn't this a hack for method symbols not considering metadata location for its range? Seems like what's proposed in #38134 is the same fix.

**Review by IMS94 — CHANGES_REQUESTED** [review]

File: compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFinder.java (hunk 1 above)
Comment (dulajdilshan) [reply compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFinder.java:1267]: We are going to track theses changes in this PR. Will be closing the #38134

Comment (dulajdilshan) [reply compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFinder.java:1267]: We have done the same kind of thing for class-fields as well 
https://github.com/ballerina-platform/ballerina-lang/blob/32a3f8efa6b2cfa79ad4848fa25ced86923c3f0a/compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFinder.java#L1260-L1262

This PR will address https://github.com/ballerina-platform/ballerina-lang/issues/37381 and https://github.com/ballerina-platform/ballerina-lang/issues/36351 but not https://github.com/ballerina-platform/ballerina-lang/issues/38053

**Review by dulajdilshan — COMMENTED** [review]
Let's add visiting annot-attachments in NodeFinder under visit(function)

**Review by dulajdilshan — APPROVED** [review]
LGTM

**Review by sanjana — APPROVED** [review]
LGTM

**Pull request merged**